library/src/main/java/com/google/maps/android/data/Renderer.java

```diff
@@ -94,9 +96,30 @@
 
     private HashMap<KmlGroundOverlay, GroundOverlay> mGroundOverlayMap;
 
-    private final ArrayList<String> mMarkerIconUrls;
+    private final Set<String> mMarkerIconUrls;
+
+    /**
+     * Map of image URL to map of scale factor to BitmapDescriptors for point marker icons
+     *
+     * BitmapDescriptors are cached to avoid creating new BitmapDescriptors for each individual
+     * usage of a Bitmap. Each BitmapDescriptor copies the Bitmap it's created from.
+     */
+    private Map<String, Map<Double, BitmapDescriptor>> mMarkerImagesCache;
 
-    private LruCache<String, Bitmap> mImagesCache;
+    /**
+     * Map of image URL to BitmapDescriptors for non-scaled ground overlay images
+     */
+    private Map<String, BitmapDescriptor> mGroundOverlayImagesCache;
+
+    /**
+     * Map of image URL to Bitmap
+     *
+     * Holds initial references to bitmaps so they can be scaled and BitmapDescriptors cached.
+     * This cache is cleared once all icon URLs are loaded, scaled, and cached as BitmapDescriptors.
+     */
+    private Map<String, Bitmap> mBitmapCache;
```

Comment:
Suggestion: thoughts on wrapping `mMarkerImagesCache`,  `mGroundOverlayImagesCache` and `mBitmapCache` into a class? I think this would help with readability and simplify the API a bit.

Reply:
Good call. Done: https://github.com/googlemaps/android-maps-utils/pull/609/commits/e9d8f24609e3055861f9961ffcde5f3fe5a09485